cmd/docker-app/status.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"github.com/deislabs/duffle/pkg/action"
+	"github.com/deislabs/duffle/pkg/claim"
+	"github.com/deislabs/duffle/pkg/credentials"
+	"github.com/deislabs/duffle/pkg/utils/crud"
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/spf13/cobra"
+)
+
+func statusCmd(dockerCli command.Cli) *cobra.Command {
+	var opts credentialOptions
+
+	cmd := &cobra.Command{
+		Use:   "status <installation-name>",
+		Short: "Get the installation status. If the installation is a docker application, the status shows the stack services.",
+		Args:  cli.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runStatus(dockerCli, args[0], opts)
+		},
+	}
+	opts.addFlags(cmd.Flags())
+
+	return cmd
+}
+
+func runStatus(dockerCli command.Cli, claimName string, opts credentialOptions) error {
+	muteDockerCli(dockerCli)
+	h := duffleHome()
+
+	claimStore := claim.NewClaimStore(crud.NewFileSystemStore(h.Claims(), "json"))
+	c, err := claimStore.Read(claimName)
+	if err != nil {
+		return err
+	}
+	targetContext := getTargetContext(opts.targetContext)
+
+	driverImpl, err := prepareDriver(dockerCli)
+	if err != nil {
+		return err
+	}
+	creds, err := prepareCredentialSet(targetContext, dockerCli.ContextStore(), c.Bundle, opts.credentialsets)
+	if err != nil {
+		return err
+	}
+	if err := credentials.Validate(creds, c.Bundle.Credentials); err != nil {
+		return err
+	}
```

Comment:
Most of the prefix of this function (to here) seems to be common with the other commands, is there an opportunity to refactor?

Reply:
Tried to refactor, but it ends with something like this:
```go
func prepareAction(dockerCli command.Cli, claimName string, opts credentialOptions) (*claim.Store, *claim.Claim, map[string]string, driver.Driver, error) {
	muteDockerCli(dockerCli)
	h := duffleHome()

	claimStore := claim.NewClaimStore(crud.NewFileSystemStore(h.Claims(), "json"))
	c, err := claimStore.Read(claimName)
	if err != nil {
		return nil, nil, nil, nil, err
	}
...
```
And at the end it can just be applied to `status` and `uninstall` commands.

So I think there no much value with this refactor. I promise I tried 😄